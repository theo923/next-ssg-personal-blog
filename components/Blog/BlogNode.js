import React from 'react';
import Link from 'next/link';


const BlogNode = ({ data, post }) => {
    return (
        <div data-test="component-blog-blogNode">
            <Link id={post} href={'/blog/'+ post}>
                <div className="my-4 w-full px-2 py-4 md:mx-2 bg-white cursor-pointer hover:bg-gray-200 rounded-lg">

                    <div className="my-2">
                        <h2 className="font-prozaLibre text-2xl font-bold text-gray-700  hover:text-gray-600 ">{data?.title}</h2>
                        <h3 className="font-spartan italic text-md mt-2 text-gray-500 ">{data?.description}</h3>
                    </div>
                    
                    <div className="flex items-center justify-between">
                        <div className='flex'>
                            {!data?.tags ? null : data.tags.map(tag => <a href='#' key={tag} className="font-spartan italic mr-2 pl-2 py-1 pt-2 pr-3 text-xs font-bold text-gray-100 transition-colors duration-200 transform bg-gray-600 rounded cursor-pointer">{tag}</a>)}
                        </div>
                        <span className="font-spartan text-md font-light text-purple-800 ">{data?.date}</span>
                    </div>
                </div>
            </Link>
        </div>
    )
}

export default BlogNode;
