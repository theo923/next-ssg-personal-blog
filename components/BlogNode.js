import React from 'react';
import Link from 'next/link';


const BlogNode = ({ data, post }) => {
    return (
        <div>
            <Link id={post} href={'/blog/'+ post}>
                <div className="my-4 w-full px-8 py-4 mx-2 bg-white cursor-pointer hover:bg-gray-200 rounded-lg dark:bg-gray-800">

                    <div className="my-2">
                        <p className="font-prozaLibre text-2xl font-bold text-gray-700 dark:text-white hover:text-gray-600 dark:hover:text-gray-200">{data?.title}</p>
                        <p className="font-spartan italic text-md mt-2 text-gray-600 dark:text-gray-300">{data?.description}</p>
                    </div>
                    
                    <div className="flex items-center justify-between">
                        <div className='flex'>
                            {!data?.tags ? null : data.tags.map(tag => <a href='#' key={tag} className="font-spartan italic mr-2 pl-2 py-1 pt-2 pr-3 text-xs font-bold text-gray-100 transition-colors duration-200 transform bg-gray-600 rounded cursor-pointer">{tag}</a>)}
                        </div>
                        <span className="text-sm font-light text-gray-600 dark:text-gray-400">{data?.date}</span>
                    </div>
                </div>
            </Link>
        </div>
    )
}

export default BlogNode;
