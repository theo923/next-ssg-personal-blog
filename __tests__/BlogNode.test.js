import { shallow } from 'enzyme';
import React from 'react';
import BlogNode from '../components/BlogNode';

describe('BlogNode test', () => {
    const mockBlogNodeData = {}

    it("expect BlogNode component is rendered without crashing", () => {
        const wrapper = shallow(<BlogNode mockBlogNodeData={mockBlogNodeData} />);   
    });

    it("expect BlogNode component is rendered && match with the Snapshot", () => {
        const wrapper = shallow(<BlogNode mockBlogNodeData={mockBlogNodeData} />);   
        expect(shallow(<BlogNode mockBlogNodeData={mockBlogNodeData} />)).toMatchSnapshot();
    });
    
});

