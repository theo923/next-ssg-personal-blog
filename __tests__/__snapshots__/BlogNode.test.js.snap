// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogNode test expect BlogNode component is rendered && match with the Snapshot 1`] = `
<div>
  <Link
    href="/blog/undefined"
  >
    <div
      className="my-4 w-full px-8 py-4 mx-2 bg-white cursor-pointer hover:bg-gray-200 rounded-lg dark:bg-gray-800"
    >
      <div
        className="my-2"
      >
        <p
          className="font-prozaLibre text-2xl font-bold text-gray-700 dark:text-white hover:text-gray-600 dark:hover:text-gray-200"
        />
        <p
          className="font-spartan italic text-md mt-2 text-gray-600 dark:text-gray-300"
        />
      </div>
      <div
        className="flex items-center justify-between"
      >
        <div
          className="flex"
        />
        <span
          className="text-sm font-light text-gray-600 dark:text-gray-400"
        />
      </div>
    </div>
  </Link>
</div>
`;
